dvc/scm/git/__init__.py

```diff
@@ -217,17 +217,28 @@ def list_tags(self):
 
     def _install_hook(self, name, cmd):
         command = "dvc {}".format(cmd)
```

Comment:
We could to it like this and avoid dvc_present_check and fobj.write() additions down below :) 
```suggestion
        command = "[ -z "$(git ls-files .dvc)" ] && exit 0\n"
        command += "dvc {}".format(cmd)
```

Reply:
Done, except:

* make it a oneliner with `||` to avoid the `exit 0`, because that's unfriendly to others who might append to the hook script (like dvc itself does if the script already exists)
* the `fobj.write()` call still needed to be changed because of the `exec`